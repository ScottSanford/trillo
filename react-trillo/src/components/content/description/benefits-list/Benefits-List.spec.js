import React from 'react'
import { shallow } from 'enzyme'
import BenefitsList from './Benefits-List'

describe('Benefits List Component', () => {

	it('should render correctly with no props', () => {
		const wrapper = shallow(<BenefitsList />)
		expect(wrapper).toMatchSnapshot()
	})

	it('should render the benefits list of 2 items', () => {
		const list = ['one', 'two']
		const wrapper = shallow(<BenefitsList benefits={list} />)

		expect(wrapper.find('.list__item')).toHaveLength(2)
		expect(wrapper.find('.list__item').at(0).props().children).toEqual('one')
		expect(wrapper).toMatchSnapshot()
	})
	
	it('should render 0 benefits if none are passed', () => {
		const list = []
		const wrapper = shallow(<BenefitsList benefits={list} />)

		expect(wrapper.find('.list__item')).toHaveLength(0)
		expect(wrapper).toMatchSnapshot()
	})
})
