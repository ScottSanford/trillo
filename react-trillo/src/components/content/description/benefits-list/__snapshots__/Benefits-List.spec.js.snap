// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Benefits List Component should render 0 benefits if none are passed 1`] = `
<ul
  className="list"
/>
`;

exports[`Benefits List Component should render correctly with no props 1`] = `""`;

exports[`Benefits List Component should render the benefits list of 2 items 1`] = `
<ul
  className="list"
>
  <li
    className="list__item"
    key="0"
  >
    one
  </li>
  <li
    className="list__item"
    key="1"
  >
    two
  </li>
</ul>
`;
